lib/spack/spack/cmd/external.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from __future__ import print_function
+from collections import defaultdict, namedtuple
+import os
+import re
+import six
+
+import spack
+import llnl.util.tty as tty
+import spack.util.spack_yaml as syaml
+
+description = "add external packages to Spack configuration"
+section = "config"
+level = "short"
+
+
+def setup_parser(subparser):
+    sp = subparser.add_subparsers(
+        metavar='SUBCOMMAND', dest='external_command')
+
+    sp.add_parser('find', help=external_find.__doc__)
+
+
+def is_executable(path):
+    return os.path.isfile(path) and os.access(path, os.X_OK)
+
+
+def _get_system_executables():
+    """Get the paths of all executables available from the current PATH.
+
+    For convenience, this is constructed as a dictionary where the keys are
+    the executable paths and the values are the names of the executables
+    (i.e. the basename of the executable path).
+
+    There may be multiple paths with the same basename. In this case it is
+    assumed there are two different instances of the executable.
+    """
+    path = os.getenv('PATH')
+    search_paths = list(p for p in path.split(os.pathsep) if os.path.isdir(p))
```

Comment:
We can use functions already in other modules:
```suggestion
    hints = spack.util.environment.get_path('PATH')
    search_paths = llnl.util.filesystem..search_paths_for_executables(*hints)
```